Add info icon and tooltip for image rating categories

CTGitWhiz opened the pull request:
This commit adds an info icon next to the image and implements a tooltip functionality that displays brief descriptions of the five best image evaluation/rating categories when the user clicks on the icon. Each category is clearly labeled with a name to help the user better understand how to evaluate the image. The tooltip disappears when the user clicks outside of it. This feature enhances user experience by providing them with more information on how to evaluate the image.

--> [US](https://github.com/users/CTGitWhiz/projects/1/views/1?pane=issue&itemId=22734482)

CTGitWhiz committed:
Add info icon and tooltip for image rating categories

CTGitWhiz committed:
Add comments

stefankorder commented on pages/image/[id].js:
```diff
@@ -126,9 +190,45 @@ const ImagePage = () => {
+        >
+          <InfoIcon />
+          <TooltipText show={showTooltip}>
+            <strong>Creativity:</strong>
+            <br />
+            Measures the uniqueness and originality of the AI-generated image.
+            <br />
+            <br />
+            <strong>Technical Quality:</strong>
+            <br /> Evaluates the level of detail, clarity, and overall
+            craftsmanship of the image.
+            <br />
+            <br />
+            <strong>Emotional Impact:</strong>
+            <br /> Considers how well the image captures and conveys an
+            emotional tone or atmosphere.
+            <br />
+            <br />
+            <strong>Coherence:</strong>
+            <br /> Determines how well the various elements of the image fit
+            together harmoniously.
+            <br />
+            <br />
+            <strong>Realism:</strong>
+            <br /> Gauges how accurately and convincingly the image represents
+            its subject matter.
+            <br />
```
Please use css to create space and no `<br />` tags. You could use a `headline` instead of a `strong` tag and a `p` tag for each part of the rest of your text.

CTGitWhiz replied:
In this updated code, I've created a new TooltipTitle component for the headings and a TooltipTextItem component for the text paragraphs. I've also used CSS properties to add spacing between the elements and change the font size. Finally, I've removed the `<br> `tags and instead used appropriate HTML tags (`<h2>` for headings and `<p>` for paragraphs) to structure the content of the tooltip.

--> [CODE](https://github.com/CTGitWhiz/capstone-project/pull/11/commits/12fcb2a8898a0f91410c464a2757bfe60fd07825)

Thanks :D

stefankorder reviewed (changes requested):
Looks good so far. But there is one important point you should change. Take a look at my comment.

CTGitWhiz committed:
Refactor tooltip with styled components

stefankorder reviewed (approved):
👍🏻

CTGitWhiz committed:
Improved Tooltip and Mobile-View on Firefox

CTGitWhiz commented:
[Commit]([Improved Tooltip and Mobile-View on Firefox](https://github.com/CTGitWhiz/capstone-project/pull/11/commits/87ba4cd484b4b26ddfb832c448075619ea3a8ef6))

Description:
Updated the implementation of the tooltip to make it more accessible, especially for users on mobile devices (Firefox-Browser). The tooltip can now be triggered using the "active" state instead of "hover", making it work for both touch and click events. Also, moved the tooltip position slightly upwards so that it does not hide the icon. Additionally, added the "focus" and "blur" events and set the "tabIndex" attribute to enable keyboard navigation for users who rely on assistive technologies. Reduced the font size for the tooltip text items and titles to improve readability.
